### PR TITLE
client update to angular 16

### PR DESCRIPTION
**Pull request by msmannan00**
1. Angular updated to angular 16
2. libraries updated to support latest angular
3. older client moved to legacy folder

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,66 @@
+backend/env
```
Comment (evilaliv3) [inline .gitignore:1]: @msmannan00 would you please remove this file and squash the commit?

**Comment by msmannan00** [comment]
> @msmannan00 : For the moment i suggest to replace this patch and just have a console.log() where we use `document.querySelector` to log something simple like "Selector not found: $selectorname".
> 
> The reason is that we don't know which could be the effect of unloading the CSS/Javascript of the user; the triggered code will have anyhow already applied changes to the page that we could not unload.
> 
> What do you think?

its just a suggestion to implement route reloading without refreshing the page if a JavaScript error occurs, and avoid loading the JavaScript this time. This approach would ensure that nothing gets messed up. Otherwise, if we don't adopt this strategy, numerous issues could arise when we release the code. Users might attempt to use the code despite the errors, leading to undesirable outcomes

**Comment by evilaliv3** [comment]
> its just a suggestion to implement route reloading without refreshing the page if a JavaScript error occurs, and avoid loading the JavaScript this time. This approach would ensure that nothing gets messed up. Otherwise, if we don't adopt this strategy, numerous issues could arise when we release the code. Users might attempt to use the code despite the errors, leading to undesirable outcomes


i see.
thank you for your feedback @msmannan00 

At least for the initial release i suggest anyhow to avoid to include this patch and think to it a little further.
Lets add eventually just a log about the found selectors.

I really think in fact that:
- removing a loaded javascript will not rever the changes that the javascript will have already applied.
- removing a javascript at the first error will prevent users to possiblly identify which are the parts that do not function; with a log on the not-found selectors we could collect feedback to fix the custmization

**Comment by msmannan00** [comment]
> > its just a suggestion to implement route reloading without refreshing the page if a JavaScript error occurs, and avoid loading the JavaScript this time. This approach would ensure that nothing gets messed up. Otherwise, if we don't adopt this strategy, numerous issues could arise when we release the code. Users might attempt to use the code despite the errors, leading to undesirable outcomes
> 
> 
> i see.
> thank you for your feedback @msmannan00 
> 
> At least for the initial release i suggest anyhow to avoid to include this patch and think to it a little further.
> Lets add eventually just a log about the found selectors.
> 
> I really think in fact that:
> - removing a loaded javascript will not rever the changes that the javascript will have already applied.
> - removing a javascript at the first error will prevent users to possiblly identify which are the parts that do not function; with a log on the not-found selectors we could collect feedback to fix the custmization

Agreed, for this release will make changes to remove the revert functionality while keeping selector logs, user would be able to see error logs and make changes accordingly for their scripts. As discussed earlier we can also mention it in application change log for globaleaks 5

**Comment by evilaliv3** [comment]
Closing since all the code is now integrated in branch main and released in version 5.0.0

**Pull request closed without merging**